addons/purchase_requisition/models/purchase_requisition.py

```diff
@@ -111,32 +122,24 @@ def action_cancel(self):
             requisition.purchase_ids.button_cancel()
             for po in requisition.purchase_ids:
                 po.message_post(body=_('Cancelled by the agreement associated to this quotation.'))
-        self.write({'state': 'cancel'})
+        self.state = 'cancel'
 
     def action_in_progress(self):
```

Comment:
Talked to thomas directly: the state (type/name) can be changed to `"confirmed, 'Confirmed'"` since this will be clearer for Blanket Orders (i.e. it will be more accurate than the current state name when it's before/after the validity dates) and it won't matter if this state shows for confirmed Purchase Templates (in the list view)